smarts/core/controllers/imitation_controller.py

```diff
@@ -53,19 +53,14 @@ def perform_action(cls, dt, vehicle, action):
         target_heading = (vehicle.heading + angular_velocity * dt) % (2 * math.pi)
 
         if isinstance(chassis, BoxChassis):
-            speed_x, speed_y = radians_to_vec(vehicle.heading) * vehicle.speed
-            new_position = np.array(
-                (
-                    vehicle.position[0] + dt * speed_x,
-                    vehicle.position[1] + dt * speed_y,
-                    vehicle.position[2],
-                )
-            )
-            target_speed = vehicle.speed + acceleration * dt
+            heading_vec = radians_to_vec(vehicle.heading)
+            ortho_vec = radians_to_vec(vehicle.heading + math.pi / 2)
+            dpos = (heading_vec * vehicle.speed + ortho_vec * angular_velocity) * dt
             new_pose = Pose(
+                position=vehicle.position + np.append(dpos, 0.0),
                 orientation=fast_quaternion_from_angle(target_heading),
-                position=new_position,
             )
+            target_speed = vehicle.speed + acceleration * dt
```

Comment:
These changes are also for a different reason. This is mainly a simplification/cleanup of the code to make it easier to read, but also not take `angular_velocity` into account for position updates.